### DISCUSSION TITLE
farcaster devnet

### DISCUSSION THREAD

**Discussion by vrypan** [farcasterxyz/protocol#134] [FIP Stage 1: Ideas]
Title: farcaster devnet\
Type: Standards FIP\
Author: Panagiotis Vryonis\
Status: Early Draft

# Abstract
Defines a devnet for farcaster.

# Problem
Developers don't have a devnet where to try early versions of their code and experiment freely (and without cost for onchain transactions). An additional issue is that the current storage limits are too high for developers to test how their apps will behave when users reach them.

# Specification

The farcaster devnet is a farcaster network with slightly different configuration:

1. Smart contracts are deployed on OP TESTNET.
2. Storage Units expire after 7 days.
3. Storage Units have limited capacity (100 casts, 100 reactions, 20 links, 10 verifications, 10 user data)
4. Hubs use different ports 3281-3283
5. Messages use FARCASTER_NETWORK_DEVNET

Hubble will have to use this configuration when `--network=3`

Farcaster clients are encouraged to provide a devnet configuration option that allows other developers to user them on the devnet.

**Comment by varunsrin** [comment]
We have a testnet today, but haven't seen much usage from developers. Will close for now and re-open if there is a lot more usage in the future.
